asllib/doc/ASLSemanticsReference.tex

```diff
@@ -455,52 +2206,42 @@ \section{SemanticsRule.Binop \label{sec:SemanticsRule.Binop}}
   \VerbatimInput[firstline=\BinopBegin, lastline=\BinopEnd]{../Interpreter.ml}
 
 \begin{formal}
-  \subsection{Formally: sequential case}
-
-  \begin{align}
-  \interp{\texttt{e1} + \texttt{e2}} (\texttt{env}) & \triangleq
-    \left\{ (v, \texttt{new\_env}) \st{}
-    \begin{aligned}
-      & \texttt{v1} + \texttt{v2} = \texttt{v}
-      \\ \text{and}\ &
-      (\texttt{v1}, \texttt{env'}) \in{} \interp{\texttt{e1}} (E)
-      \\ \text{and}\ &
-      (\texttt{v2}, \texttt{new\_env}) \in{} \interp{\texttt{e2}} (\texttt{env'})
-    \end{aligned}
-    \right\}
-  \label{eq:sem-seq-plus}
-  \end{align}
-
-  \subsection{Formally: concurrent case}
-  For the ordering constraints, the different arguments of a n-ary subprogram
-are considered computed in parallel:
-  \begin{align}
-    \interp{\texttt{e1} + \texttt{e2}} (\texttt{env}) & \triangleq
-    \left\{ (\texttt{v}, \texttt{new\_env}, S_1 \parallel S_2) \st{}
-    \begin{aligned}
-      & \texttt{v1} + \texttt{v2} = \texttt{v}
-      \\ \text{and}\ &
-      (\texttt{v1}, \texttt{env'}, S_1) \in{} \interp{\texttt{e1}} (\texttt{env})
-      \\ \text{and}\ &
-      (\texttt{v2}, \texttt{new\_env}, S_2) \in{} \interp{\texttt{e2}} (\texttt{env'})
-    \end{aligned}
-    \right\}
-  \label{eq:sem-conc-plus}
-  \end{align}
+  \subsection{Formally}
+  \begin{mathpar}
+    \inferrule{
+      \evalexpr{ \env, \veone} \evalarrow \Normal(\vmone, \envp) \OrAbnormal \\
+      \evalexpr{ \envp, \vetwo } \evalarrow \Normal(\vmtwo, \newenv) \OrAbnormal \\
+      \vmone = (\vvone, \vgone) \\
+      \vmtwo = (\vvtwo, \vgtwo) \\
+      \binop(\op, \vvone, \vvtwo) \evalarrow \vv \terminateas \ErrorConfig\\
+      \vg = \vgone \parallel \vgtwo
+    }
+    {
+      \evalexpr{ \env, \texttt{E\_Binop}(\op, \veone, \vetwo) } \evalarrow
+      \Normal((\vv, \vg), \newenv)
+    }
+  \end{mathpar}
 \end{formal}
 
 \subsection{Comments}
-  This is related to \identr{BKNT}.
+  This is related to \identr{XKGC} and \identr{BKNT}.
 
-\section{SemanticsRule.Unop \label{sec:SemanticsRule.Unop}}
+Notice that when one of the sub-expressions terminates exceptionally,
+the other exception is guaranteed by the type-checker to be side effect-free and non-throwing (\identi{QJTN}).
+
```

Comment:
please refer to the correct place in the ASLTypingReference, not QJTN.

Reply:
I rephrased this as
> Notice that when one of the sub-expressions terminates exceptionally,
> the other expression must be side effect-free and non-throwing (\identi{QJTN}).
The type-checker doesn't yet actually check this.